pkg/kubectl/cmd/cmd.go

```diff
@@ -385,6 +391,20 @@ func printDeprecationWarning(command, alias string) {
 	glog.Warningf("%s is DEPRECATED and will be removed in a future version. Use %s instead.", alias, command)
 }
 
+// deprecatedCmd is intended to be used to create a "wrapper" command around an
+// existing command. The wrapper works the same but prints a deprecation
+// message before running.
+func deprecatedCmd(deprecatedVersion string, cmd *cobra.Command) *cobra.Command {
+	// Have to be careful here because Cobra automatically extracts the name
+	// of the command from the .Use field.
+	originalName := cmd.Name()
+
+	cmd.Use = deprecatedVersion
+	cmd.Deprecated = fmt.Sprintf("use %q instead", originalName)
+	cmd.Hidden = true
+	return cmd
+}
+
```

Comment:
was thinking more about the unit test for this function. We can write a unit test which wraps a NoOp cobra.Command and captures the "deprecated" string in the output. Thoughts ? (happy to discuss in person)